exp/api/v1alpha3/azuremachinepool_types.go

```diff
@@ -68,6 +73,23 @@ type (
 		SpotVMOptions *infrav1.SpotVMOptions `json:"spotVMOptions,omitempty"`
 	}
 
+	// AzureDefaultingImage defines information about the image to use for VMSS VM creation.
+	// There are three ways to specify an image: by ID, Marketplace Image or SharedImageGallery
+	// One of ID, SharedImage or Marketplace should be set.
+	//
+	// If no image is specified, the image will be defaulted by the controller and Defaulted set to true which will
+	// inform the controller that if the parent MachinePool.Spec.Template.Spec.Version field is updated, this image
+	// will be updated with the corresponding default image. If Defaulted is set to false, the controller will not
+	// update the image reference when the MachinePool.Spec.Template.Spec.Version changes.
+	AzureDefaultingImage struct {
```

Comment:
Thought we should be explicit about image defaulting. This enables users to use default image versions for K8s versions specified on MachinePools while being safe to specify their own without updates to MachinePools overriding their image reference.

Reply:
This sounds like a good idea. Should these `AzureDefaultingImage` changes be part of a different PR instead?